MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/MovingIconWithTrailingLineActivity.java

```diff
@@ -0,0 +1,309 @@
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconAllowOverlap;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconIgnorePlacement;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconImage;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconOffset;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconSize;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.lineCap;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.lineColor;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.lineJoin;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.lineWidth;
+
+/**
+ * Make a directions request with the Mapbox Directions API and then draw a line behind a moving
+ * SymbolLayer icon which moves along the Directions response route.
+ */
+public class MovingIconWithTrailingLineActivity extends AppCompatActivity {
+
+  private static final String DOT_SOURCE_ID = "dot-source-id";
+  private static final String LINE_SOURCE_ID = "line-source-id";
+
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+
+  private GeoJsonSource pointSource;
+  private GeoJsonSource lineSource;
+  private List<Point> routeCoordinateList;
+  private List<Point> markerLinePointList = new ArrayList<>();
+  private int routeIndex;
+
+  private Animator currentAnimator;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_dds_moving_icon_with_trailing_line);
+
+    // Initialize the mapboxMap view
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(MapboxMap mapboxMap) {
+        MovingIconWithTrailingLineActivity.this.mapboxMap = mapboxMap;
+        mapboxMap.setStyle(Style.LIGHT, new Style.OnStyleLoaded() {
+          @Override
+          public void onStyleLoaded(@NonNull Style style) {
+            // Use the Mapbox Directions API to get a directions route
+            getRoute(style,
+              Point.fromLngLat(38.7508, 9.0309), // coffee shop
+              Point.fromLngLat(38.795902, 8.984467) // airport
+            );
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * Add data to the map once the GeoJSON has been loaded
+   *
+   * @param featureCollection returned GeoJSON FeatureCollection from the Directions API route request
+   */
+  private void initData(@NonNull FeatureCollection featureCollection) {
```

Comment:
The `Style` object can be passed to `initData`.

Reply:
Fixed